File: codility/cpp/codility_lesson7_3.cpp
// you can use includes, for example:
// #include <algorithm>
#include <stack>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(string &S) {
	// write your code in C++14 (g++ 6.2.0)
	if (S.empty())
		return 1;

	stack<char> s;
	int N = S.size();

	for (int i = 0; i < N; i++)
	{
		if (S[i] == '(')
			s.push(S[i]);
		else
		{
			if (s.empty())
				return 0;
			else
				s.pop();
		}
	}

	if (s.empty())
		return 1;
	else
		return 0;
}

File: codility/python/codility_lesson7_3.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(S):
	# write your code in Python 3.6
	ans = 1
	stack = []
	for s in S:
		if s == "(":
			stack.append(s)
		else:
			if len(stack) != 0:
				stack.pop()
			else:
				ans = 0
				break
	if len(stack) != 0:
		ans = 0
	return ans
